pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseCombineOperator.java

```diff
@@ -153,7 +154,18 @@ public void runJob() {
    */
   protected void processSegments(int threadIndex) {
     for (int operatorIndex = threadIndex; operatorIndex < _numOperators; operatorIndex += _numThreads) {
-      IntermediateResultsBlock resultsBlock = (IntermediateResultsBlock) _operators.get(operatorIndex).nextBlock();
+      Operator operator = _operators.get(operatorIndex);
+      IntermediateResultsBlock resultsBlock;
+      try {
+        if (operator instanceof AcquireReleaseColumnsSegmentOperator) {
+          ((AcquireReleaseColumnsSegmentOperator) operator).acquire();
+        }
```

Comment:
I think this should be outside of the try block, so that we don't release the operator if acquire fails. Same for other places

Reply:
actually i intentionally put it inside the try, so that we release no matter what happens in acquire or execution

Reply:
following up to our offline discussion about it not following lock/unlock semantics, will stick to this for now and revisit if need be, since functionality wise it's the same